Missionframework/modules/24_enemy/fnc/fn_enemy_spawnPatrol.sqf

```diff
@@ -0,0 +1,110 @@
+/*
+    KPLIB_fnc_enemy_spawnPatrol
+
+    File: fn_enemy_spawnPatrol.sqf
+    Author: KP Liberation Dev Team - https://github.com/KillahPotatoes
+    Date: 2019-06-08
+    Last Update: 2019-06-15
+    License: GNU General Public License v3.0 - https://www.gnu.org/licenses/gpl-3.0.html
+    Public: Yes
+
+    Description:
+        Spawns a (motorized) patrol at nearest sector of given position. Random sector if no position is given.
+        Destination of the patrol will be nearest enemy sector.
+
+    Parameter(s):
+        _spawnPos   - Position look for nearest sector to spawn patrol from. If empty, random sector selection  [ARRAY, defaults to []]
+        _motorized  - Bool to select if the patrol should be motorized                                          [BOOL, defaults to false]
+        _unitCount  - Amount of infantry units, when not motorized                                              [NUMBER, defaults to 4]
+
+    Returns:
+        Spawned patrol [GROUP]
+*/
+
+params [
+    ["_spawnPos", [], [[]]],
+    ["_motorized", false, [false]],
+    ["_unitCount", 4, [0]]
+];
+
+// Only inactive opfor sectors
+private _sectors = (KPLIB_sectors_all - KPLIB_sectors_blufor - KPLIB_sectors_active);
+private _patrolSectors = [];
+private _grp = grpNull;
+private _veh = objNull;
+
+// Get random sector, when no position given. Otherwise get nearest sector
+if (_spawnPos isEqualTo []) then {
+    _patrolSectors pushBack (_sectors deleteAt (floor random count _sectors));
+} else {
+    _patrolSectors pushBack ([1200, _spawnPos, _sectors] call KPLIB_fnc_core_getNearestMarker);
+    _sectors = _sectors - _patrolSectors;
+};
+
+// Exit if no sector was found
+if ((_patrolSectors select 0) isEqualTo "") exitWith {
+    private _text = format ["Spawn patrol couldn't find a sector within 1200m distance of given position: %1", _spawnPos];
+    [_text, "ENEMY"] call KPLIB_fnc_common_log
+};
+
+// Fetch some nearby patrol sectors
+_patrolSectors append (_sectors select {((getMarkerPos (_patrolSectors select 0)) distance2D (getMarkerPos _x)) < 2000});
+
+// Exit if sector is too far away from other sectors for a patrol route
+if ((count _patrolSectors) < 2) exitWith {
+    private _text = format ["Spawn patrol couldn't find sectors within 2000m distance for patrol from sector: %1 (%2)", markerText (_patrolSectors select 0), _patrolSectors select 0];
+    [_text, "ENEMY"] call KPLIB_fnc_common_log
+};
+
+// Take a maximum of 4 sectors
+if ((count _patrolSectors) > 4) then {_patrolSectors resize 4;};
```

Comment:
Won't this result with always having same patrol routes between sectors? Maybe always shuffle the array before this `if` ?

Only thing is that you would need to keep start sector in separate variable otherwise patrol might start not where you want it to start. Maybe in L51 store additional sectors in variable, shuffle then append.